roqoqo-derive/src/involve_qubits.rs

```diff
@@ -138,6 +162,26 @@ fn involve_qubits_struct(ds: DataStruct, ident: Ident) -> TokenStream {
                 }
             }
         }
+    } else if control_2 {
+        if !(control_0 && control_1 && target) {
+            panic!("When deriving InvolveQubits for a four-qubit operation control_0, control_1, control_2 and target have to be present");
+        };
+        // Creating a function that puts qubits `control_0`, `control_1` `control_2` and `target` into the InvolvedQubits HashSet
+        quote! {
+            /// Implements [InvolveQubits] trait for the qubits involved in this Operation.
+            #[automatically_derived]
+            impl InvolveQubits for #ident{
+                /// Returns a list of all involed qubits.
+                fn involved_qubits(&self ) -> InvolvedQubits {
+                    let mut new_hash_set: std::collections::HashSet<usize> = std::collections::HashSet::new();
+                    new_hash_set.insert(self.control_0);
+                    new_hash_set.insert(self.control_1);
+                    new_hash_set.insert(self.control_2);
+                    new_hash_set.insert(self.target);
+                    InvolvedQubits::Set(new_hash_set)
+                }
+            }
+        }
     } else if control_0 || control_1 {
```

Comment:
```suggestion
    } else if control_0 || control_1 || control_2 {
```
And adapt the panic message

Reply:
`control_2` is already checked previously for the 4-qubit operations. This one is for 3-qubit ones.

Reply:
The previous check checks that if there is control_2 there is also `control_0`, `control_1` and `target` but shouldn't there also be a test to ensure that there is not `control_2` and `control` ?